src/java.base/share/classes/sun/security/provider/certpath/CertId.java

```diff
@@ -172,19 +172,13 @@ public void encode(DerOutputStream out) {
     }
 
     /**
-     * Returns a hashcode value for this CertId.
-     *
-     * @return the hashcode value.
+     * {@return a hashcode value for this CertId}
      */
     @Override public int hashCode() {
         if (myhash == -1) {
             myhash = hashAlgId.hashCode();
-            for (int i = 0; i < issuerNameHash.length; i++) {
-                myhash += issuerNameHash[i] * i;
-            }
-            for (int i = 0; i < issuerKeyHash.length; i++) {
-                myhash += issuerKeyHash[i] * i;
-            }
+            myhash += Arrays.hashCode(issuerNameHash);
+            myhash += Arrays.hashCode(issuerKeyHash);
             myhash += certSerialNumber.getNumber().hashCode();
```

Comment:
It seems that for some other classes, you may just use Objects.hash(hashAlgId, issuerNameHash, issuerKeyHash, certSerialNumber.getNumber())? Is there a rule on what to use?

Reply:
Hmm, I see. Thanks for the clarification.